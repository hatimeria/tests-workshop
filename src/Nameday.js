const Nameday = () => {
    return (
        <div>
            <button>See your namedays</button>
        </div>
    )
}

export default Nameday;