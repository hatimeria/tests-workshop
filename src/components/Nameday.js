import { useState } from 'react';
import axios from 'axios';

const Nameday = ({ name }) => {
  const [nameDays, setNameDays] = useState([]);
  const [error, setError] = useState(false);
  const getNameday = async () => {
    try {
      const response = await axios.get(`https://nameday.abalin.net/api/V1/getname`, {
        params: {
          name,
          country: 'pl'
        }
      });
      setNameDays(response.data[0])
      setError(false);
    } catch(e) {
      console.error(e);
      setError(true);
    }
  }

  return (
    <div>
      {name && <button onClick={() => getNameday()}>See your namedays</button>}
      {error && <p>Sth went wrong, try again</p>}
      {(nameDays && nameDays.length) ? (
        <div>
          <h2>Your namedays are:</h2>
          <ul>
            {nameDays.map(el => (
              <li key={`${el.day}-${el.month}`}>
                Day: {el.day}, Month: {el.month}
              </li>
            ))}
          </ul>
        </div>
      ) : ''}
    </div>
  )
}

export default Nameday;