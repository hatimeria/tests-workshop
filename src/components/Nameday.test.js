import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Nameday from './Nameday';

test('renders button', () => {
  render(<Nameday name="Mieszko" />);
  const button = screen.getByRole('button', { name: 'See your namedays' });
  expect(button).toBeInTheDocument();
});

test('doesnt render button if name wasnt passed', () => {
  render(<Nameday name="" />);
  const button = screen.queryByRole('button', { name: 'See your namedays' });
  expect(button).not.toBeInTheDocument();
});

test('nameday is displayed when the button is clicked', async () => {
  render(<Nameday name="Igor" />);
  const button = screen.getByRole('button', { name: 'See your namedays' });
  fireEvent.click(button);
  const list = await screen.findByRole('list');
  expect(list).toBeInTheDocument();

  const namedays = await screen.findAllByRole('listitem');
  expect(namedays).toHaveLength(2);
});

test('renders correctly', () => {
  const tree = renderer
    .create(<Nameday name="CzÄ™stogniew" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});