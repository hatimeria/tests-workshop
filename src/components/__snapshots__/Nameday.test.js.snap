// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    See your namedays
  </button>
  0
</div>
`;
